src/common/options/global.yaml.in

```diff
@@ -4610,6 +4610,73 @@ options:
   flags:
   - runtime
   with_legacy: true
+- name: bluestore_kv_sync_interval
+  type: size
+  level: dev
+  long_desc: Bluestore syncronizes key/value data and deferred writes to the RocksDB
+    write-ahead-log in a dedicated background thread.  This paramater specifies in
+    microseconds how long the kv sync thread should wait to accumulate IO before
+    flushing data the the write-ahead-log.  The higher this value is set, the longer
+    the kv sync thread will wait.  This can increase minimum latency, but may also
+    reduce the number of syncs necessary to flush data to disk.  The effect on
+    average and tail latency are likely to be device dependent. A value of 0 means
+    not to wait and let the kv_sync thread run as fast as possible.
+  default: 0
+  flags:
+  - runtime
+  with_legacy: true
+- name: bluestore_kv_sync_interval_hdd
+  type: size
+  level: dev
+  default: 8000
+  see_also:
+  - bluestore_kv_sync_interval
+  flags:
+  - runtime
+  with_legacy: true
+- name: bluestore_kv_sync_interval_ssd
+  type: size
+  level: dev
+  default: 0
+  see_also:
+  - bluestore_kv_sync_interval
+  flags:
+  - runtime
+  with_legacy: true
+- name: bluestore_kv_sync_items
```

Comment:
For HDDs, isn't the primary motivation to reduce head movement?  IE it seems to me that we almost don't care about the type or size of an op until they get very large (which should be bounded by the time) but rather try to aggregate as many ops, whatever they are, as possible.  I confess I haven't looked closely at how kv_throttle_costs is calculated.

Reply:
well, perhaps you're right and we should care about op count rather than cost. On the other hand piling "costly" ops isn't good too - this will trigger throttling and might cause over ops stalling...